src/coreclr/jit/importer.cpp

```diff
@@ -3834,84 +3803,66 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                                 bool                    tailCall,
                                 CORINFO_RESOLVED_TOKEN* pConstrainedResolvedToken,
                                 CORINFO_THIS_TRANSFORM  constraintCallThisTransform,
-                                CorInfoIntrinsics*      pIntrinsicID,
                                 NamedIntrinsic*         pIntrinsicName,
                                 bool*                   isSpecialIntrinsic)
 {
-    assert((methodFlags & (CORINFO_FLG_INTRINSIC | CORINFO_FLG_JIT_INTRINSIC)) != 0);
+    assert((methodFlags & CORINFO_FLG_INTRINSIC) != 0);
 
-    bool              mustExpand  = false;
-    bool              isSpecial   = false;
-    CorInfoIntrinsics intrinsicID = CORINFO_INTRINSIC_Illegal;
-    NamedIntrinsic    ni          = NI_Illegal;
+    bool           mustExpand  = false;
+    bool           isSpecial   = false;
+    NamedIntrinsic ni          = NI_Illegal;
 
     if ((methodFlags & CORINFO_FLG_INTRINSIC) != 0)
-    {
-        intrinsicID = info.compCompHnd->getIntrinsicID(method, &mustExpand);
-
-#ifdef DEBUG
-        if (intrinsicID != CORINFO_INTRINSIC_Illegal)
-        {
-            JITDUMP("Intrinsic %s Recognized\n", impGetIntrinsicName(intrinsicID));
-        }
-#endif // DEBUG
-    }
-
-    if ((methodFlags & CORINFO_FLG_JIT_INTRINSIC) != 0)
     {
         // The recursive non-virtual calls to Jit intrinsics are must-expand by convention.
         mustExpand = mustExpand || (gtIsRecursiveCall(method) && !(methodFlags & CORINFO_FLG_VIRTUAL));
```

Comment:
@jkotas, this is the `mustExpand` for all `NamedIntrinsic`.

Reply:
I do not see how this marks the `ByReference<T>` intrinsics as must-expand. I think there may need to be an explicit check, similar to `NI_System_StubHelpers_GetStubContext` or `NI_System_Runtime_CompilerServices_RuntimeHelpers_CreateSpan` below

Reply:
Yup, fixed by https://github.com/dotnet/runtime/pull/62639/commits/03cd05fa04463e75f1c94bed671c660f04ad2809, that was the root cause of sigsegv. :slightly_smiling_face: